cyclonedx.go

```diff
@@ -266,6 +268,268 @@ type Credits struct {
 	Individuals   *[]OrganizationalContact `json:"individuals,omitempty" xml:"individuals>individual,omitempty"`
 }
 
+type AssetType string
+
+const (
+	AssetTypeAlgorithm             AssetType = "algorithm"
+	AssetTypeCertificate           AssetType = "certificate"
+	AssetTypeProtocol              AssetType = "protocol"
+	AssetTypeRelatedCryptoMaterial AssetType = "related-crypto-material"
+)
+
+type Primitive string
```

Comment:
Some of these types have rather generic names (e.g. `Primitive`, `Mode`). We need to pay attention that we won't cause naming collisions as the spec grows. Consider prefixing some of these types so they're "pseudo-namespaced". For example `CryptoPrimitive` instead of `Primitive`.

Reply:
Yup, I see your point and I agree. I think `Crypto` works as a prefix. I will work on that.